util/dvsim/Scheduler.py

```diff
@@ -233,44 +104,372 @@ def on_sigint(signal_received, frame):
         finally:
             signal(SIGINT, old_handler)
 
-        # We got to the end without anything exploding. Return the results for our jobs.
+        # We got to the end without anything exploding. Return the results.
         return self.item_to_status
 
+    def _schedule_items(self, items):
+        '''Convert a heterogenous list of items to run into list of tasks
+        segregated by target and cfg (DUT).'''
```

Comment:
It might be cleaner to promote the nested function (`add_to_scheduled`) to be a proper method to match `_remove_from_scheduled` and then inline the top-level loop into the call site at `__init__`. Then there are no nested functions (always nice) and things look a bit more symmetrical.

If you don't want to do that, please make this a static method and also fix the docstring style.